activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -671,6 +671,29 @@ def remove_reference(table_name, ref_name, options = {})
       def foreign_keys(table_name)
         raise NotImplementedError, "foreign_keys is not implemented"
       end
+      
+      # Checks to see if a foreign key exists on a table for a given foreign key definition.
+      #
+      #   # Check a foreign key exists
+      #   foreign_key_exists?(:accounts, :branches)
+      #
+      #   # Check a foreign key on specified column exists
+      #   foreign_key_exists?(:accounts, column: :owner_id)
+      #
+      #   # Check a foreign key with a custom name exists
+      #   foreign_key_exists?(:accounts, name: "special_fk_name")
+      #
+      def foreign_key_exists?(from_table, options_or_to_table = {})
```

Comment:
I would consider renaming to `foreign_key_exist?` to avoid any confusion. For example `File.exists?` in Ruby is deprecated in favor of `File.exist?`. Should we follow that convention?